sign.go

```diff
@@ -0,0 +1,374 @@
+package cose
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+// signature represents a COSE_Signature CBOR object:
+//
+//   COSE_Signature =  [
+//       Headers,
+//       signature : bstr
+//   ]
+//
+// Reference: https://tools.ietf.org/html/rfc8152#section-4.1
+type signature struct {
+	_           struct{} `cbor:",toarray"`
+	Protected   cbor.RawMessage
+	Unprotected cbor.RawMessage
+	Signature   []byte
+}
+
+// Signature represents a decoded COSE_Signature.
+//
+// Reference: https://tools.ietf.org/html/rfc8152#section-4.1
+type Signature struct {
+	Headers   Headers
+	External  []byte
```

Comment:
COSE_Signature does not have an external data property. I also recall having this conversation and saying that there should only be on external data per signed message, regardless of the signature count.

Reply:
I agree with @qmuntal here!

Reply:
@qmuntal Could you create an issue for this so that we can address later to unblock this PR.